src/resources/JsResolver.ts

```diff
@@ -18,8 +19,48 @@ export class JsResolver {
       );
     }
 
-    const requestTemplateContent = fs.readFileSync(this.config.path, 'utf8');
-    return this.processTemplateSubstitutions(requestTemplateContent);
+    return this.processTemplateSubstitutions(this.getResolverContent());
+  }
+
+  getResolverContent(): string {
+    if (this.api.config.esbuild === false) {
+      return fs.readFileSync(this.config.path, 'utf8');
+    }
+
+    // process with esbuild
+    // this will:
+    // - Bundle the code into one file if necessary
+    // - Transpile typescript to javascript if necessary
+    const buildResult = buildSync({
+      target: 'esnext',
+      sourcemap: 'inline',
+      treeShaking: true,
+      minifyWhitespace: true,
+      minifyIdentifiers: true,
```

Comment:
```suggestion
      minifyIdentifiers: false,
```
Otherwise it converts: `import{runtime}from\"@aws-appsync/utils\";` into `import{runtime as $}from\"@aws-appsync/utils\";`

Reply:
Do you have a way to reproduce this? 

In my case it is transpiled to 
```js
import{util as e,runtime as r}from\"@aws-appsync/utils\";
```

(from this)
```ts
import {
  Context,
  DynamoDBGetItemRequest,
  util,
  runtime,
} from '@aws-appsync/utils';
```

and it's working as expected at runtime.

Reply:
Sorry. You are right. I had other issues that the eslint appsync configuration was not caching and I thought it was that.

Reply:
OK, it seems like this sometimes causes issues after all.

Someone recently asked me for help with an issue on AppSync, using esbuild. 
He hit an `Runtime error` without any explanation from AppSync. 

After some debugging, we figured out he used the `--minify-identifiers` option with esbuild which was renaming a variable to `_`. 

That variable was then passed into a custom function as an argument. 
For some reason, AppSync does not like that.

Debugging in the appsync code evaluator gives this error:

```
Lexical error, Encountered: "_" (95), after : "." at *unset*[line 1, column 876]
```

example code 

```ts
    const { foo: _ } = ctx.args;

    const r = test(_)
```

It sounds like a rare use case but we might need to consider removing any minification for now (`minifyWhitespace` might be safe to keep though) to avoid any issue.

The [official AppSync doc](https://docs.aws.amazon.com/appsync/latest/devguide/resolver-reference-overview-js.html#additional-utilities) does not include modification in their guides.

Reply:
we can leave it as opt-in though thought the `esbuild` config. 
We could even consider allowing different `esbuild` options per resolver/funciton to allow fine-tuning each one and work around those kind of issues when they are found

Reply:
I'm not sure minifying make sense anyway. Is there a limit on each function size? Otherwise I would rather be able to read the whole request/response in case I need what has been deployed

Reply:
There is a limit.
4kb for inline code (which is what this plugin does at the moment), or 10kb if uploaded to S3. 

I've had feedback of people hitting the 4kb limit 😄